### ISSUE TITLE
FreeBSD CI does not work as expected

### ISSUE THREAD

**Issue by tertsdiepraam** [uutils/coreutils#4163]
Recently, FreeBSD CI was fixed, but there are some rough edges.

I first got a flaky (I think) test, but all I saw in the CI output was that it failed. Only after scrolling through all the tests was I able to find the actual error, so we need better error reporting:
``` 
 ---- test_tail::test_follow_name_truncate4 stdout ----
current_directory_resolved: 
write(append): /tmp/.tmpSLjjyw/file
run: /Users/runner/work/coreutils/coreutils/target/debug/coreutils tail -s.1 --max-unchanged-stats=1 -F file
write(truncate): /tmp/.tmpSLjjyw/file
thread 'test_tail::test_follow_name_truncate4' panicked at 'assertion failed: buf_stderr.is_empty()', tests/by-util/test_tail.rs:1985:9
```
https://github.com/uutils/coreutils/actions/runs/3503280450

Then I reran the test and got a lot of "file has vanished" and at the end
```
rsync warning: some files vanished before they could be transferred (code 24) at /AppleInternal/Library/BuildRoots/a0876c02-1788-11ed-b9c4-96898e02b808/Library/Caches/com.apple.xbs/Sources/rsync/rsync/main.c(1404) [generator=2.6.9]
```
https://github.com/uutils/coreutils/actions/runs/3503280450/jobs/5868960733

**Comment by miles170** [comment]
Check the https://github.com/uutils/coreutils/actions/runs/3503280450/jobs/5868960733 raw log.

I saw the `ssh_dispatch_run_fatal: Connection to 127.0.0.1 port 2222: message authentication code incorrect` which indicates the SSH is disconnecting while the test is still running.

And the `file has vanished: /root/work/coreutils/coreutils/target/*` is because the files were cleaned by `cargo clean` while the rsync was copying files.
The issue might be related to the broken SSH connection.

**Comment by tertsdiepraam** [comment]
Not my area of expertise to be honest. If you think it would work we can give it a shot!

**Comment by miles170** [comment]
Would you like to add this [commit](https://github.com/miles170/coreutils/commit/2c0024544c5f716d04b0d719d35a9dd08a7781d7) to your [PR](https://github.com/uutils/coreutils/pull/4158)?
I have checked that the FreeBSD CI passed. (https://github.com/miles170/coreutils/actions/runs/3506882167/jobs/5874237565)

Check similar issue here: https://github.com/vmactions/freebsd-vm/issues/28

**Comment by tertsdiepraam** [comment]
I'd prefer to fix this in a separate PR. Could you open one?
